src/rgw/const.py

```diff
@@ -96,3 +97,14 @@ class RgwEndpoint(Enum):
     MOTR_PROCESS_FID = 'motr my fid'
 
 
+# ConfStore keys of rgw.py
+AUTH_USER_KEY = f'cortx>{COMPONENT_NAME}>auth_user'
+AUTH_ADMIN_KEY = f'cortx>{COMPONENT_NAME}>auth_admin'
+AUTH_SECRET_KEY = f'cortx>{COMPONENT_NAME}>auth_secret'
+CLUSTER_ID_KEY = 'cluster>id'
+MACHINE_IDS_KEY = 'cluster>storage_set[0]>nodes'
```

Comment:
what is this key? Why only storage_set '0' is considered?

Reply:
Its intended to denote current node. But what i done here is just moved that existing  conf key ('cluster>storage_set[0]>nodes') from rgw.py to const.py so that next time if we want to make any changes in Machine ID key, value location we can simply update this constant.

Reply:
> Its intended to denote current node. 

Ok, so 0th index always gives current node, on any given node? 

> But what i done here is just moved that existing conf key

Yeah i understand this

Reply:
I think this may fail in case of multiple storage_sets.
The correct way to get total number of nodes(machine_ids) is,
```
>>> from cortx.utils.conf_store import Conf, MappedConf
>>> Conf.load('index', 'yaml:///root/cluster.conf')
>>> node_id=Conf.machine_id
# storage_set value for given node.
>>> storage_set=Conf.get('index', f'node>{node_id}>storage_set')
>>> storage_set
'storage-set-1'
>>>
>>> storage_set_count=Conf.get('index', 'cluster>storage_set_count')
# Total number of nodes for given storage set.
>>> for i in range(0, storage_set_count):
...     if Conf.get('index', f'cluster>storage_set[{i}]>name') == storage_set:
...             total_num_of_nodes_in_storage_set = Conf.get('index', f'cluster>storage_set[{i}]>nodes')
...
>>> total_num_of_nodes_in_storage_set
['d4c353b9797046f9b2ba61b3049c9c11', '5086adab8c2546a4a65672400d94fedb', 'e848f1799b834dfd9037857268d2158c', 'a94dbb84f49c                                                                                        4f079995e6343284fe68']
>>>

```

Reply:
Hi @shriya-deshmukh @cdeshmukh Above change will work as expected. 

Means, conf.get('index', 'cluster>storage_set[0]>nodes') will return list of machine_id

if it is single storage we get single machine id or if multi storage the we get multiple machine id.

And internally these Machine id list are looped to do required work. 

say below code is from rgw.py,
```
machine_ids = Rgw._get_cortx_conf(conf, const.MACHINE_IDS_KEY)
data_pod_hostnames = [Rgw._get_cortx_conf(conf, f'node>{machine_id}>hostname') for machine_id in machine_ids if Rgw._get_cortx_conf(conf, f'node>{machine_id}>type') == 'data_node']
```

sample cluster.conf

```
cluster:
  id: f256fa9ca9474a2eb0a862ea6b328321
  name: cortx-cluster
  num_storage_set: 1
  storage_set:
  - durability:
      dix:
        data: '1'
        parity: '0'
        spare: '0'
      sns:
        data: '1'
        parity: '0'
        spare: '0'
    name: storage-set-1
    nodes:
    - ac135656f80047a787e458a7e01a8cc2
    - e3944301b8d84710b145fdb31b45f35a
    - dee37000b6b1446eaff4af9b5de1d5f7
    - 87af2d35805340ffa63c66cc24641cac
    num_nodes: 4
  storage_set_count: 1
```

Reply:
Hi @cdeshmukh ​ had a call with @shriya-deshmukh ​ to discuss about multi storage set scenario which will replace this hardcoded indexed[0] way. I will create a new ticket with this requirements.